### PR TITLE
Add the strictMode option to not terminate the current ripley session if an invalid HTTP request is received

### PR DESCRIPTION
**Pull request by eugenepaniot**
Do not break session if got invalid HTTP Method

### CODE REVIEW

**Review by dwilliams782 — APPROVED** [review]

**Review by georgemalamidis-lh — COMMENTED** [review]
@eugenepaniot @dwilliams782 I've thought of this before. The reason for the `panic` here is to brute force prompt us there is bad input data so that we can take a look at it and fix it, i.e. I think it's useful for ripley to not be resilient here. If we log the error and continue instead, I'm worried we'll never know, unless we explicitly add STDERR checks wherever ripley is used.

**Comment by eugenepaniot** [comment]
> @eugenepaniot @dwilliams782 I've thought of this before. The reason for the `panic` here is to brute force prompt us there is bad input data so that we can take a look at it and fix it, i.e. I think it's useful for ripley to not be resilient here. If we log the error and continue instead, I'm worried we'll never know, unless we explicitly add STDERR checks wherever ripley is used.

If it skips incorrect req and exits with non-zero code, will it be fine? Or input data for ripley has to be properly prepared?

**Comment by georgemalamidis-lh** [comment]
> > @eugenepaniot @dwilliams782 I've thought of this before. The reason for the `panic` here is to brute force prompt us there is bad input data so that we can take a look at it and fix it, i.e. I think it's useful for ripley to not be resilient here. If we log the error and continue instead, I'm worried we'll never know, unless we explicitly add STDERR checks wherever ripley is used.
> 
> If it skips incorrect req and exits with non-zero code, will it be fine? Or input data for ripley has to be properly prepared?

That would take care of the cronjob finishing in a `Failed` state, which is good. 
I can't think of many command line tools that continue working when given bad input. Can you think of any? However, we do pass ripley a *list* of inputs, so maybe it's OK to log, have the cronjob tell us it failed and then we go and investigate the bad data. What's really important for me is to always investigate bad data and the cronjob completely blowing up forces us to do that. 
If you think log bad input + non 0 exit code is the best way to go, let's do that.

**Comment by eugenepaniot** [comment]
> > > @eugenepaniot @dwilliams782 I've thought of this before. The reason for the `panic` here is to brute force prompt us there is bad input data so that we can take a look at it and fix it, i.e. I think it's useful for ripley to not be resilient here. If we log the error and continue instead, I'm worried we'll never know, unless we explicitly add STDERR checks wherever ripley is used.
> > 
> > 
> > If it skips incorrect req and exits with non-zero code, will it be fine? Or input data for ripley has to be properly prepared?
> 
> That would take care of the cronjob finishing in a `Failed` state, which is good. I can't think of many command line tools that continue working when given bad input. Can you think of any? However, we do pass ripley a _list_ of inputs, so maybe it's OK to log, have the cronjob tell us it failed and then we go and investigate the bad data. What's really important for me is to always investigate bad data and the cronjob completely blowing up forces us to do that. If you think log bad input + non 0 exit code is the best way to go, let's do that.

For example `grep` command:
```
bash-4.4$ grep Name /proc/1/mem  /proc/1/status
grep: /proc/1/mem: Permission denied
/proc/1/status:Name:	systemd

bash-4.4$ echo $?
2
```

I think it's ok to continue working, in non-critical sections, if possible, but change exitCode.

File: main.go [hunk 1]
```diff
@@ -20,22 +20,25 @@ package main
...
+
+	ripley "github.com/loveholidays/ripley/pkg"
 )
 
 func main() {
 	paceStr := flag.String("pace", "10s@1", `[duration]@[rate], e.g. "1m@1 30s@1.5 1h@2"`)
 	silent := flag.Bool("silent", false, "Suppress output")
 	dryRun := flag.Bool("dry-run", false, "Consume input but do not send HTTP requests to targets")
 	timeout := flag.Int("timeout", 10, "HTTP client timeout in seconds")
+	strictMode := flag.Bool("strictMode", false, "Strict mode")
```
Comment (georgemalamidis-lh) [inline main.go:35]: @eugenepaniot I'd change `strictMode` to `strict` and `"Strict mode"` to `"Panic on bad input"`

File: pkg/replay.go [hunk 2]
```diff
@@ -55,9 +57,21 @@ func Replay(phasesStr string, silent, dryRun bool, timeout int) {
 
 	for scanner.Scan() {
 		req, err := unmarshalRequest(scanner.Bytes())
-
 		if err != nil {
-			panic(err)
+			if strictMode {
+				panic(err)
+			} else {
+				exitCode = 126
```
Comment (georgemalamidis-lh) [inline pkg/replay.go:64]: @eugenepaniot no need for an `else` block here, we can save two lines of and a bit of indentation.

Comment (eugenepaniot) [reply pkg/replay.go]: done

File: main.go (hunk 1 above)
Comment (eugenepaniot) [reply main.go]: done

**Pull request merged**